sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/PropertiesAccessCorrectFieldBase.cs

```diff
@@ -83,16 +84,21 @@ protected void CheckType(SymbolAnalysisContext context)
                 .OfType<IPropertySymbol>()
                 .Where(p => !p.IsImplicitlyDeclared);
 
-        protected void CheckExpectedFieldIsUsed(IFieldSymbol expectedField, FieldData? actualField, SymbolAnalysisContext context)
+        protected void CheckExpectedFieldIsUsed(IFieldSymbol expectedField, IEnumerable<FieldData?> actualFields, SymbolAnalysisContext context)
         {
-            if (actualField.HasValue && actualField.Value.Field != expectedField)
+            var expectedFieldIsUsed = actualFields.Any(a => a.HasValue && a.Value.Field == expectedField);
```

Comment:
What will the rule do if the code being tested doesn't refer to any fields? (e.g. just has a method call)